### PR TITLE
Decompile func_ptr_80170008_w_050 and func_ptr_8017000C_w_050

### PR DESCRIPTION
**Pull request by joshlory**
* [func_ptr_80170008_w_050](https://decomp.me/scratch/HoEkr)
* [func_ptr_8017000C_w_050](https://decomp.me/scratch/5k56G)

### CODE REVIEW

File: src/weapon/w_050.c [hunk 1]
```diff
@@ -192,9 +192,207 @@ void func_162000_8017BA38(Entity* ent, Point16* outPoint, bool arg2) {
...
+
+    switch (self->step) {
+    case 0:
+        self->primIndex = g_api.AllocPrimitives(PRIM_LINE_G2, PrimCount);
+        if (self->primIndex != -1) {
+
+            self->flags = FLAG_UNK_800000 | FLAG_UNK_40000 | FLAG_UNK_20000;
+            prim = &g_PrimBuf[self->primIndex];
+            for (i = 0; i < PrimCount; i++) {
+                angle = i << 0x18 >> 0x10;
```
Comment (sozud) [inline src/weapon/w_050.c:230]:
```
                angle = var_i << 8;
```
same for other instances of this

Comment (joshlory) [reply src/weapon/w_050.c:230]: Hmm I'm not getting a match with that change... can you share a scratch?

**Review by sozud — APPROVED** [review]

File: src/weapon/w_050.c [hunk 2]
```diff
@@ -192,9 +192,207 @@ void func_162000_8017BA38(Entity* ent, Point16* outPoint, bool arg2) {
...
+    offsetY = PLAYER.posY.i.hi;
+    self->posY.i.hi = offsetY - 0x1A;
+    if (PLAYER.drawFlags & FLAG_DRAW_ROTY) {
+        self->posY.i.hi = offsetY - 0x1F;
+    }
+
+    switch (self->step) {
+    case 0:
+        self->primIndex = g_api.AllocPrimitives(PRIM_LINE_G2, PrimCount);
+        if (self->primIndex != -1) {
```
Comment (Xeeynamo) [inline src/weapon/w_050.c:225]:
```suggestion
        if (self->primIndex != -1) {
```
Let's invert the condition to remove indentation: https://decomp.me/scratch/nm9HF

The `if (primIndex == -1) { DestroyEntity(self); return; }` is a pattern found in almost all the code base

File: src/weapon/w_050.c [hunk 3]
```diff
@@ -192,9 +192,207 @@ void func_162000_8017BA38(Entity* ent, Point16* outPoint, bool arg2) {
 
 INCLUDE_ASM("weapon/nonmatchings/w_050", func_ptr_80170004);
 
-INCLUDE_ASM("weapon/nonmatchings/w_050", func_ptr_80170008);
+void func_ptr_80170008(Entity* self) {
+    const int PrimCount = 16;
```
Comment (Xeeynamo) [inline src/weapon/w_050.c:196]: Nice idea!

**Review by Xeeynamo — APPROVED** [review]
:ok_hand:  with a tiny :nail_care:

File: src/weapon/w_050.c (hunk 3 above)
Comment (joshlory) [reply src/weapon/w_050.c:196]: I think you suggested it in https://github.com/Xeeynamo/sotn-decomp/pull/1401#discussion_r1673048922 — so yeah, good idea! 😄

**Pull request merged**